src/detect-pcre.c

```diff
@@ -921,10 +931,12 @@ static void DetectPcreFree(void *ptr)
 
     DetectPcreData *pd = (DetectPcreData *)ptr;
 
-    if (pd->re != NULL)
-        pcre_free(pd->re);
-    if (pd->sd != NULL)
-        pcre_free_study(pd->sd);
+    if (pd->parse_regex.regex != NULL)
```

Comment:
call a generic func to clean up the parse_regex here?

Reply:
Will do.